src/Input/StandardQuestionTrait.php

```diff
@@ -69,6 +67,29 @@ private function createQuestion(): Question
         );
     }
 
+    /**
+     * @param null|string|string[] $defaultValue
+     */
+    private function getDefaultPrompt($defaultValue): string
+    {
+        if (null === $defaultValue) {
+            return '';
+        }
+
+        if (is_array($defaultValue)) {
+            Assert::allScalar($defaultValue, 'One or more default values were not presented as scalars');
```

Comment:
`allScalar()` conflicts with `string[]` here (redundant check)

Reply:
This is a hairy one.

Essentially, symfony/console is specifying incorrect types, which is what is leading to issues, and it comes down to how it handles input from a question. All question input is passed first to a normalizer, and then to a validator. A normalizer can change the value type, and a validator is expected to return the value if it is valid (and raise an exception otherwise); these are expected to return a scalar (and, more specifically, `bool|int|string`, though there's no reason they couldn't return `float` as well). When a _single_ value is expected in return, symfony/console follows this type. However, when _multiple_ values are returned, it types the list as `string[]` — even though each individual value in the list goes through the same normalizer/validator logic, and thus can be of types other than `string`. As such, I did `allScalar()` here, but followed the symfony/console `string[]` type, because otherwise psalm produced errors.

Reply:
I was able to document this like I did in `ChoiceParam`, so all good now.